kernel/src/hil/digest.rs

```diff
@@ -6,23 +6,50 @@ use crate::ErrorCode;
 /// Implement this trait and use `set_client()` in order to receive callbacks.
 ///
 /// 'L' is the length of the 'u8' array to store the digest output.
-pub trait Client<'a, const L: usize> {
+pub trait ClientData<'a, const L: usize> {
     /// This callback is called when the data has been added to the digest
     /// engine.
     /// On error or success `data` will contain a reference to the original
     /// data supplied to `add_data()`.
     fn add_data_done(&'a self, result: Result<(), ErrorCode>, data: &'static mut [u8]);
+}
 
+/// Implement this trait and use `set_client()` in order to receive callbacks.
+///
+/// 'L' is the length of the 'u8' array to store the digest output.
+pub trait ClientHash<'a, const L: usize> {
     /// This callback is called when a digest is computed.
     /// On error or success `digest` will contain a reference to the original
     /// data supplied to `run()`.
     fn hash_done(&'a self, result: Result<(), ErrorCode>, digest: &'static mut [u8; L]);
 }
 
+/// Implement this trait and use `set_client()` in order to receive callbacks.
+///
+/// 'L' is the length of the 'u8' array to store the digest output.
+pub trait ClientVerify<'a, const L: usize> {
+    /// This callback is called when a verification is computed.
+    /// On error or success `digest` will contain a reference to the original
+    /// data supplied to `verify()`.
+    /// On success the result indicate if the hashes match or don't.
+    /// On failure the result will indicate an `ErrorCode`.
+    fn verification_done(&'a self, result: Result<bool, ErrorCode>, compare: &'static mut [u8; L]);
+}
+
+pub trait Client<'a, const L: usize>:
+    ClientData<'a, L> + ClientHash<'a, L> + ClientVerify<'a, L>
+{
+}
+
```

Comment:
It would also make sense have standard composite traits for Data+Verify and Data+Hash? `ClientDataHash` and `ClientDataVerify`. Those seem like the two most common use cases for capsules.